From 8b2101f96d7aff09f19194aeaf0bb4f421f201a5 Mon Sep 17 00:00:00 2001
From: sukanka <su975853527@gmail.com>
Date: Wed, 18 Jun 2025 00:59:12 +0800
Subject: [PATCH 1/2] fix for boost 1.88

`assign` has been depreciated in favor of `fill` in boost 1.88, see the [changelog](https://www.boost.org/doc/libs/latest/libs/array/doc/html/array.html#reference_modifiers)

1	1	pkg/dem/TriaxialStressController.hpp

diff --git a/pkg/dem/TriaxialStressController.hpp b/pkg/dem/TriaxialStressController.hpp
index 146eb54..f89a9c3 100644
--- a/pkg/dem/TriaxialStressController.hpp
+++ b/pkg/dem/TriaxialStressController.hpp
@@ -130,7 +130,7 @@ public:
 		strain = Vector3r::Zero();
    		first = true;
 		stiffness.resize(6);
-		previousTranslation.assign(Vector3r::Zero());
+		previousTranslation.fill(Vector3r::Zero());
 		for (int i=0; i<6; ++i){normal[i]=stress[i]=force[i]=Vector3r::Zero();stiffness[i]=0;}
 		normal[wall_bottom].y()=1;
 		normal[wall_top].y()=-1;
-- 
2.50.0

